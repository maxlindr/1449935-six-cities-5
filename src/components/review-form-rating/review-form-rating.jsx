import React from 'react';
import PropTypes from 'prop-types';

const starLabelTitle = {
  5: `perfect`,
  4: `good`,
  3: `not bad`,
  2: `badly`,
  1: `terribly`
};

function ReviewFormRating(props) {
  const {rating, onClick, disabled} = props;

  return (
    <div className="reviews__rating-form form__rating">
      {Array(5).fill().map((val, i) => i + 1).reverse().map((starValue) => (
        <React.Fragment key={starValue}>
          <input
            className="form__rating-input visually-hidden"
            name="rating"
            id={`${starValue}-stars`}
            type="radio"
            defaultValue={starValue}
            checked={rating === starValue}
            onChange={onClick}
            disabled={disabled}
          />

          <label
            htmlFor={`${starValue}-stars`}
            className="reviews__rating-label form__rating-label"
            title={starLabelTitle[starValue]}>
            <svg className="form__star-image" width={37} height={33}>
              <use xlinkHref="#icon-star" />
            </svg>
          </label>
        </React.Fragment>
      ))}
    </div>
  );
}

ReviewFormRating.propTypes = {
  rating: PropTypes.number,
  disabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
};

export default React.memo(ReviewFormRating);
