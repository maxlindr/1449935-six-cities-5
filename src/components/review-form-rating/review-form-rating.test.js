import React from 'react';
import renderer from 'react-test-renderer';
import {ReviewFormRating} from './review-form-rating';

test(`ReviewFormRating должен корректно отображаться`, () => {
  const component = renderer.create(
      <ReviewFormRating
        rating={3}
        disabled={false}
        onClick={() => {}}
      />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
