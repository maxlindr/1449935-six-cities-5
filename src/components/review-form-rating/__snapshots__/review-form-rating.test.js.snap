// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewFormRating должен корректно отображаться 1`] = `
<div
  className="reviews__rating-form form__rating"
>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    defaultValue={5}
    disabled={false}
    id="5-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height={33}
      width={37}
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    defaultValue={4}
    disabled={false}
    id="4-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="4-stars"
    title="good"
  >
    <svg
      className="form__star-image"
      height={33}
      width={37}
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={true}
    className="form__rating-input visually-hidden"
    defaultValue={3}
    disabled={false}
    id="3-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="3-stars"
    title="not bad"
  >
    <svg
      className="form__star-image"
      height={33}
      width={37}
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    defaultValue={2}
    disabled={false}
    id="2-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="2-stars"
    title="badly"
  >
    <svg
      className="form__star-image"
      height={33}
      width={37}
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    defaultValue={1}
    disabled={false}
    id="1-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="1-stars"
    title="terribly"
  >
    <svg
      className="form__star-image"
      height={33}
      width={37}
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
</div>
`;
